### DISCUSSION TITLE
Expected time to next block should be adjusted based on current block speed

### DISCUSSION THREAD

**Discussion by softsimon** [mempool/mempool#563] [Ideas]
Here is an illustration of how it could look like, with -20% slower blocks than expected, it is in instead 12 minutes until next block (on average).

<img width="883" alt="Screen Shot 2021-05-13 at 15 56 02" src="https://user-images.githubusercontent.com/8561090/118122465-f13adb80-b403-11eb-8d67-9f62111ca474.png">

**Comment by RequestPrivacy** [comment]
I just wanted to open an issue but here you are :D Glad you brought it up, that would be a really neat feature. Also cool would be a dynamic estimated block time, so that it counts the minutes down. At the time where the block should be (on average) mined, a label could indicate the pending mining.
